include/pqxx/transaction_base.hxx

```diff
@@ -287,6 +257,52 @@ public:
     return r[0].as<TYPE>();
   }
 
+  /// Execute a query, and loop over the results row by row.
+  /** Converts the rows to @c std::tuple, of the column types you specify.
+   *
+   * Use this with a range-based "for" loop.  It executes the query, and
+   * directly maps the resulting rows onto a @c std::tuple of the types you
+   * specify.  It starts before all the data from the server is in, so if your
+   * network connection to the server breaks while you're iterating, you'll get
+   * an exception partway through.
+   *
+   * The tuple may contain @c std::string_view fields, but the strings to which
+   * they point will only remain valid until you extract the next row.  After
+   * that, the memory holding the string may be overwritten or deallocated.
+   *
+   * If any of the columns can be null, and the C++ type to which it translates
+   * does not have a null value, wrap the type in @c std::optional (or if
+   * you prefer, @c std::shared_ptr or @c std::unique_ptr).  These templates
+   * do recognise null values, and libpqxx will know how to convert to them.
+   *
+   * The connection is in a special state until the iteration finishes.  So if
+   * it does not finish due to a @c break or a @c return or an exception, then
+   * the entire connection becomes effectively unusable.
+   *
+   * Querying in this way is faster than the @c exec() methods, at least for
+   * larger results, and you can start processing rows before the full result
+   * is in.  Also, @c stream() scales better in terms of memory usage.  Where
+   * @c exec() reads the entire result into memory at once, @c stream() will
+   * read and process one row at at a time.
+   *
+   * Your query executes as part of a COPY command, not as a stand-alone query,
+   * so there are limitations to what you can do in the query.  It can be
+   * either a SELECT or VALUES query; or an INSERT, UPDATE, or DELETE with a
+   * RETURNING clause.  See the documentation for PostgreSQL's COPY command for
+   * the details:
+   *
+   *     https://www.postgresql.org/docs/current/sql-copy.html
+   *
+   * Iterating in this way does require each of the field types you pass to be
+   * default-constructible, copy-constructible, and assignable.  These
+   * requirements may be loosened once libpqxx moves on to C++20.
+   */
+  template<typename... TYPE>[[nodiscard]] auto stream(std::string_view query)
```

Comment:
Kind of unrelated to the changes here, but seeing that you take a std::string_view as the query text made me think that there's room here to support a constexpr data type, such as a boost::hana::string, which would let the library parse the query at compile time, and pre-compute details like parameter positions and type conversions.

Reply:
It'd save me a string allocation and some copying, which is always welcome.

I'm guessing I wouldn't be able to use the exact same code for those string types, otherwise I could just template everything on "pass whatever string type you like" and not introduce a dependency.  We've got so many string types in there already!